### PR TITLE
chore: make readfile/readParsable unaware of file descriptors

### PR DESCRIPTION
**Pull request by nulltoken**
**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

Handling file descriptors is a CLI only concern. This change makes
those two core functions unlearn how to deal with them.

### CODE REVIEW

File: src/cli/services/linter/linter.ts [hunk 1]
```diff
@@ -29,36 +32,29 @@ export async function lint(documents: Array<number | string>, flags: ILintConfig
     });
   }
 
-  if (flags.verbose) {
-    if (ruleset) {
-      const rules = Object.values(spectral.rules);
-      console.info(`Found ${rules.length} rules (${rules.filter(rule => rule.enabled).length} enabled)`);
-    } else {
-      console.info('No rules loaded, attempting to detect document type');
```
Comment (nulltoken) [inline src/cli/services/linter/linter.ts:37]: @P0lip I've dropped this as ruleset seems to always be defined.

**Comment by P0lip** [comment]
Hey, I'll take a look at it tomorrow!

File: src/cli/services/linter/linter.ts [hunk 2]
```diff
@@ -18,7 +21,7 @@ export async function lint(documents: Array<number | string>, flags: ILintConfig
   for (const [format, lookup, prettyName] of KNOWN_FORMATS) {
     spectral.registerFormat(format, document => {
       if (lookup(document)) {
-        if (!flags.quiet) {
+        if (!(flags.quiet === true)) {
```
Comment (P0lip) [inline src/cli/services/linter/linter.ts:24]:
```suggestion
        if (flags.quiet !== true) {
```

File: src/cli/services/linter/utils/segregateEntriesPerKind.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export const segregateEntriesPerKind = (entries: Array<string | number>): [string[], number[]] => {
+  return entries.reduce(
```
Comment (P0lip) [inline src/cli/services/linter/utils/segregateEntriesPerKind.ts:2]:
```suggestion
  return entries.reduce<[string[], number[]]>(
```
Would this work?

File: src/cli/services/linter/utils/segregateEntriesPerKind.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    (group, entry) => {
+      if (typeof entry === 'string') {
+        group[0].push(entry);
+        return group;
+      }
+
+      group[1].push(entry);
+      return group;
+    },
+    [[], []] as [string[], number[]],
```
Comment (P0lip) [inline src/cli/services/linter/utils/segregateEntriesPerKind.ts:12]:
```suggestion
    [[], []],
```

**Review by P0lip — APPROVED** [review]
Looks good!
To be frank, I don't think this is a breaking change.
It's an internal API, so we should be good.

**Pull request merged**